Modules/Common/PolkadotLikeMinimalModule.php

```diff
@@ -0,0 +1,240 @@
+                }
+            }
+            else
+            {
+                $this_i = 0;
+            }
+
+            if (!isset($this_i))
+                continue;  // Block 12831012 contains 0xa36adc0ddedb4b9f9f8de5ef59fa6c99bbb8ca8bbc44d189dd0bedf6bf61b92e
+                           // which is present in 12857725, 12856380, 12855736, and many more, but not in 12831012 ¯\_(ツ)_/¯
+
+            $initiating_address = $i[$this_i]['account_id'];
+            $fee_found = false;
+            $extrinsic_failed = $i[$this_i]['failed'];
+
+            foreach ($i[$this_i]['event'] as $event)
+            {
+                if ($event['event_id'] === 'TransactionFeePaid')
+                {
+                    if ($fee_found)
+                        throw new ModuleError('Two fees');
+
+                    $fee_data = json_decode($event['params'], true);
+
+                    if (count($fee_data) !== 3)
+                        throw new ModuleError('Wrong fee data');
+                    if ($fee_data[1]['type_name'] !== 'BalanceOf')
+                        throw new ModuleError('Wrong fee data');
+
+                    $fee = $fee_data[1]['value'];
+                    $fee_found = true;
+                    $fee_idx = $event['event_idx'];
+                }
+            }
+
+            if (!$fee_found) // Uh-oh
+            {
+                $fee = $i[$this_i]['fee'];
+                    $fee_idx = $i[$this_i]['event'][0]['event_idx'];
+            }
+
+            if (!$i[$this_i]['transfers'])
+                continue;
+
+            foreach ($i[$this_i]['transfers'] as $this_transfer)
+            {
+                if ((int)$this_transfer['block_num'] !== $block_id)
+                    throw new ModuleError('Wrong block');
+
+                if ($extrinsic_failed && !$this_transfer['failed'])
+                    throw new ModuleError('Not failed');
+
+                $transfers[] =
+                    ['index' => $this_transfer['event_idx'],
+                     'hash' => $this_transfer['hash'],
+                     'from' => $this_transfer['from'],
+                     'to' => $this_transfer['to'],
+                     'amount' => $this_transfer['amount'],
+                     'failed' => $this_transfer['failed'],
+                     'type' => 'transfer'
+                    ];
+            }
+
+            $transfers[] =
+                ['index' => $fee_idx,
+                 'hash' => $this_transfer['hash'],
+                 'from' => $initiating_address,
+                 'to' => 'the-void',
+                 'amount' => $fee,
+                 'failed' => false,
+                 'type' => 'fee'
+                ];
+        }
+
+        usort($transfers, function($a, $b) {
+            return [$a['index']] <=> [$b['index']];
+        });
+
+        $sort_key = 0;
+
+        foreach ($transfers as $transfer)
+        {
+            $events[] = [
+                'transaction' => $transfer['hash'],
```

Comment:
again, this is not a unique identifier, use `block_number-extrinsic_number` instead, it can also be interpreted as a string.